eole/config/models.py

```diff
@@ -323,6 +323,10 @@ class TransformerDecoderConfig(TransformerConfig, DecoderConfig):
         default=0.0,
         description="Lambda value for alignement loss of Garg et al, 2019 " "(https://arxiv.org/abs/1909.02074)",
     )
+    prefix_LM: bool = Field(
+        default=False,
+        description="Whether the TransformerDecoder is a prefixLM or CausalLM https://arxiv.org/pdf/2308.06912",
+    )
```

Comment:
Maybe something like `LM_type: Literal["prefix", "causal"]` would be more generic for potential additional logic down the line.